packages/slo-stat-bot/src/slo-stat-label.ts

```diff
@@ -0,0 +1,221 @@
+        pull_number,
+        100
+      );
+      let fileList: PullsListFilesResponseItem[] | undefined;
+
+      if (fileResponse !== undefined) fileList = fileResponse.data;
+
+      for (
+        let i = 0;
+        fileList !== undefined && fileList[i] !== undefined;
+        i++
+      ) {
+        const file = fileList[i];
+        if (
+          file.filename === '.github/issue_slo_rules.json' ||
+          (repo === '.github' && file.filename === 'issue_slo_rules.json')
+        ) {
+          await handler.handle_slos(
+            context,
+            owner,
+            repo,
+            pull_number,
+            file.sha
+          );
+          break;
+        }
+      }
+    }
+  );
+}
+
+handler.handle_slos = async function handle_slos(
+  context: Context,
+  owner: string,
+  repo: string,
+  issue_number: number,
+  file_sha: string
+) {
+  const sloString = await handler.getFileContents(
+    context.github,
+    owner,
+    repo,
+    file_sha
+  );
+
+  if (sloString !== undefined) {
+    const sloData = JSON.parse(sloString);
+    const res: ValidationResults = await handler.lint(schema, sloData);
+
+    await handler.commentPR(
+      context.github,
+      owner,
+      repo,
+      issue_number,
+      res.isValid
+    );
+    await handler.createCheck(context, res);
+  }
+};
+
+handler.getFileContents = async function getFileContents(
+  github: GitHubAPI,
+  owner: string,
+  repo: string,
+  file_sha: string
+): Promise<string | undefined> {
+  try {
+    const blob = await github.git.getBlob({
+      owner,
+      repo,
+      file_sha,
+    });
+    const fileContent = Buffer.from(blob.data.content, 'base64').toString(
+      'utf8'
+    );
+    return fileContent;
+  } catch (err) {
+    console.log(err);
+    return;
+  }
+};
+
+handler.listFiles = async function listFiles(
+  github: GitHubAPI,
+  owner: string,
+  repo: string,
+  pull_number: number,
+  per_page: number
+): Promise<Response<PullsListFilesResponse> | undefined> {
+  try {
+    const listOfFiles = await github.pulls.listFiles({
+      owner,
+      repo,
+      pull_number,
+      per_page,
+    });
+    return listOfFiles;
+  } catch (err) {
+    console.log(err);
+    return;
+  }
+};
+
+handler.lint = async function lint(
+  schema: JSON,
+  sloData: JSON
+): Promise<ValidationResults> {
+  const ajv = new Ajv();
+  const validate = await ajv.compile(schema);
+  const isValid = await validate(sloData);
+
+  return {
+    isValid: isValid,
+    errors: validate.errors,
+  } as ValidationResults;
+};
+
+handler.commentPR = async function commentPR(
+  github: GitHubAPI,
+  owner: string,
+  repo: string,
+  issue_number: number,
+  isValid: boolean
+) {
+  if (!isValid) {
+    const body =
+      'ERROR: "issue_slo_rules.json" file is not valid with JSON schema';
+    try {
+      await github.issues.createComment({
+        owner,
+        repo,
+        issue_number,
+        body,
+      });
+    } catch (err) {
+      console.log(err);
```

Comment:
@bcoe @sofisl seeing these make me a littler nervous.  What's our general guidance on exceptions?  Is it bad to "let it throw" in probot apps?

Reply:
I catch errors in some of the other bots in the same way. However, I think you want to make sure you're not catching an error that should stop execution, i.e., that you're preserving the functionality of what the bot is supposed to do. So, for example, with getFileContents(), I would make sure you catch the error and make sure you're returning what you want to check for. Later on, you check that function for undefined, so maybe make sure that in the function, if you get an error, you also return ```undefined```. Also, since this bot is supposed to get stats, make sure that you're doing something with that error message, in other words, make sure an error with that function doesn't just get lost in the logs. One idea could be that you return a file content string that has a special value that you can then run stats on to check any errors, i.e., ```return fileContent = 'error in getting contents'```, or for your listFiles response, you can return a ```PullsListFilesResponse``` item that returns a special value you can check for any errors. Or, you can also let the error stop the execution if it's essential to the bot. Just some ideas, I think it ultimately depends on how you want the bot to function!